pkg/querier/frontend/frontend.go

```diff
@@ -426,3 +434,23 @@ FindQueue:
 	// and wait for more requests.
 	goto FindQueue
 }
+
+// CheckReady determines if the query frontend is ready.  Function parameters/return
+// chosen to match the same method in the ingester
+func (f *Frontend) CheckReady(_ context.Context) error {
+	// if the downstream url is configured the query frontend is not aware of the state
+	//  of the queriers and is therefore always ready
+	if f.cfg.DownstreamURL != "" {
+		return nil
+	}
+
+	// if we have more than one querier connected we will consider ourselves ready
+	connectedClients := f.connectedClients.Load()
+	if connectedClients > 0 {
+		return nil
+	}
+
+	msg := fmt.Sprintf("not ready: connected clients %d", connectedClients)
+	level.Warn(f.log).Log("msg", msg)
```

Comment:
Warn or Info? I would prefer to avoid having warnings logged on each startup for a legit case (considering we'll have health checks done before queriers will be attached).

Reply:
agreed.